extensions/slack/src/shared/utils.ts

```diff
@@ -3,10 +3,15 @@ import formatDistance from "date-fns/formatDistance";
 import { runAppleScript } from "run-applescript";
 
 // https://api.slack.com/reference/deep-linking
-const openChat = (workspaceId: string, userId: string) => {
-  const { closeRightSidebar } = getPreferenceValues<{ closeRightSidebar: boolean }>();
+const openChat = (workspaceId: string, userId: string, conversationId: string) => {
+  const { closeRightSidebar, openInBrowser } =
+    getPreferenceValues<{ closeRightSidebar: boolean; openInBrowser: boolean }>();
 
-  open(`slack://user?team=${workspaceId}&id=${userId}`);
+  open(
+    openInBrowser
+      ? `https://app.slack.com/client/${workspaceId}/${conversationId}`
+      : `slack://user?team=${workspaceId}&id=${userId}`
+  );
   closeMainWindow();
   if (closeRightSidebar) {
```

Comment:
With the `openInBrowser` option active, closing the sidebar focus the Slack app after opening the Slack website in a browser.

We need to check for `closeRightSidebar && !openInBrowser`.

Reply:
Good catch! Resolved in followup.